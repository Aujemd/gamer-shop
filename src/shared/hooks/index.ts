export { useStoredCart } from "./use-stored-cart";
