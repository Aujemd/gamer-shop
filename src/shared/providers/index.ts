export * from "./data-fetching-provider";
