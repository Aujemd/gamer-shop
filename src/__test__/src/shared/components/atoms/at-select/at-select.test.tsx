import { fireEvent, render, screen } from "@testing-library/react";

import { AtSelect } from "@/shared";

describe("atSelect Component", () => {
    it("renders the select element", () => {
        const { asFragment } = render(
            <AtSelect>
                <option value="test-1">test 1</option>
                <option value="test-2">test 2</option>
            </AtSelect>,
        );
        expect(asFragment).toMatchSnapshot();
    });

    it("renders the children options", () => {
        render(
            <AtSelect>
                <option value="test-1">test 1</option>
                <option value="test-2">test 2</option>
            </AtSelect>,
        );
        expect(screen.getByText("test 1")).toBeInTheDocument();
        expect(screen.getByText("test 2")).toBeInTheDocument();
    });

    it("applies custom className", () => {
        render(
            <AtSelect
                data-testid="at-select-test-id"
                className="custom-class"
            >
                <option value="x">X</option>
            </AtSelect>,
        );
        const select = screen.getByTestId("at-select-test-id");

        expect(select).toHaveClass("custom-class");
    });

    it("calls onChange when an option is selected", () => {
        const handleChange = jest.fn();
        render(
            <AtSelect
                data-testid="at-select-test-id"
                onChange={handleChange}
            >
                <option value="test">test</option>
            </AtSelect>,
        );

        const select = screen.getByTestId("at-select-test-id");
        fireEvent.change(select, { target: { value: "test" } });
        expect(handleChange).toHaveBeenCalledTimes(1);
        expect(handleChange).toHaveBeenCalledWith(
            expect.objectContaining({
                target: expect.objectContaining({
                    value: "test",
                }),
            }),
        );
    });
});
