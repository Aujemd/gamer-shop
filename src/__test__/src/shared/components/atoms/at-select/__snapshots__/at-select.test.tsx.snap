// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`atSelect Component renders the select element 1`] = `[Function]`;
